src/components/Pagination/index.jsx

```diff
@@ -21,19 +22,28 @@ const Pagination = ({
     siblingCount,
     pageSize,
   });
+  const { handlePaginationQueryParamsChange } = usePaginationQueryParams();
+
+  if (!navigate) navigate = handlePaginationQueryParamsChange;
 
   if (pageNo === 0 || paginationRange.length < 2) {
     return null;
   }
 
   const onNext = () => {
-    !isLastPage && navigate(pageNo + 1);
+    if (isLastPage) return;
+
+    navigate(pageNo + 1);
```

Comment:
This is more concise:
```suggestion
    if (!isLastPage) navigate(pageNo + 1);
```